Add config flow to compensation helper (#33723)
home-assistant/home-assistant.github.io: source/_integrations/compensation.markdown

ha_codeowners:
  - '@Petro31'
ha_domain: compensation
ha_config_flow: true
ha_platforms:
  - sensor
ha_integration_type: helper
---

The **Compensation** {% term integration %} consumes the {% term state %} from other {% term sensors %}. It exports the compensated value as state in a separate {% term entity %} and the following values as attributes: `entity_id` and `coefficients`. A single polynomial, linear by default, is fit to all data points provided.

{% include integrations/config_flow.md %}

> Further information about these configuration options can be found under the [YAML configuration](#yaml-configuration)

Reviewer: ```suggestion
Further information about these configuration options can be found under the [YAML configuration](#yaml-configuration) section.
```

{% configuration_basic %}
Name:
  description: The name the sensor should have.
Entity:
  description: The entity that provides the input.
Data points:
  description: "The collection of data point conversions with the format `uncompensated_value, compensated_value`.  e.g., `1.0, 2.1`."
Attribute:
  description: Attribute from the source to monitor/compensate. When omitted, the state value of the source will be used.
Degree:

>   description: "The degree of a polynomial. e.g., Linear compensation (y = x + 3) has 1 degree, Quadratic compensation (y = x<sup>2</sup> + x + 3) has 2 degrees, etc."

Reviewer: ```suggestion
  description: "The degree of a polynomial. For example, Linear compensation (y = x + 3) has 1 degree, Quadratic compensation (y = x<sup>2</sup> + x + 3) has 2 degrees, etc."
```

Precision:

>   description: Defines the precision of the calculated values, through the argument of round().

Reviewer: Could you add an example for this one, maybe?

Author: I am rewriting it to make it less complicated so even without an example it should be understandable I think.

>   description: Defines the precision of the calculated values, through the argument of round().

Author: ```suggestion
  description: Defines the number of decimal places of the calculated sensor value.
```

Unit of measurement:
  description: Defines the units of measurement of the sensor, if any.
Lower limit:

>   description: "Enables a lower limit for the sensor. The lower limit is defined by the data collections (`data_points`) lowest `uncompensated_value`."

Reviewer: ```suggestion
  description: "Enables a lower limit for the sensor. The lower limit is defined by the data collection's (`data_points`) lowest `uncompensated_value`."
```

Upper limit:

>   description: "Enables an upper limit for the sensor. The upper limit is defined by the data collections (`data_points`) greatest `uncompensated_value`."

Reviewer: ```suggestion
  description: "Enables an upper limit for the sensor. The upper limit is defined by the data collection's (`data_points`) greatest `uncompensated_value`."
```
